src/utils/date.ts

```diff
@@ -94,3 +96,62 @@ export function nextMonth(date: Date): Date {
   }
   return nextDate;
 }
+
+/**
+ * Translate a number into a given locals numeral system
+ */
+export function localizeNumber(num: number, localeData: DateLocaleData): string {
+  return String(num)
+    .split("")
+    .map((i) => localeData.numerals[i])
+    .join("");
+}
+
+/**
+ * Calculate actual number from localized string
+ */
+export function parseNumber(str: string, localeData: DateLocaleData): number {
+  const numerals = "0123456789";
+  return parseInt(
+    str
+      .split("")
+      .map((i) => numerals[localeData.numerals.indexOf(i)])
+      .filter((num) => num)
+      .join("")
+  );
+}
+
+/**
+ * Parse numeric units for day, month, and year from a localized string
+ * month starts at 0 (can pass to date constructor)
+ */
+export function parseDateString(str: string, localeData: DateLocaleData): { day: number; month: number; year: number } {
```

Comment:
Might be nice to have the return type of the function be declared with a name like the `unitOrderSignifier` type down below.